libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c

```diff
@@ -1628,3 +1666,615 @@ TEST( MQTT_Unit_API, KeepAliveJobCleanup )
+
+/**
+ * @brief Tests that IotMqtt_SerializePingReq works as intented.
+ * to @ref mqtt_function_serializepingreq.
+ */
+TEST( MQTT_Unit_API, SerializePingReqChecks )
+{
+    uint8_t buffer[ 10 ];
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+
+    /* Buffer size less than disconnect request fails. */
+    status = IotMqtt_SerializePingreq( buffer, 1 );
+    TEST_ASSERT_EQUAL_INT( IOT_MQTT_BAD_PARAMETER, status );
+
+    /* NULL buffer fails. */
+    status = IotMqtt_SerializePingreq( NULL, 10 );
+    TEST_ASSERT_EQUAL_INT( IOT_MQTT_BAD_PARAMETER, status );
+
+    /* Good case succeeds. */
+    status = IotMqtt_SerializePingreq( buffer, 2 );
+    TEST_ASSERT_EQUAL_INT( IOT_MQTT_SUCCESS, status );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Tests that IotMqtt_GetIncomingMQTTPacketTypeAndLength works as intented.
+ * to @ref mqtt_function_getincomingmqttpackettypeandlength.
+ */
+TEST( MQTT_Unit_API, GetIncomingMQTTPacketTypeAndLengthChecks )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+    IotMqttPacketInfo_t mqttPacket;
+    uint8_t buffer[ 10 ];
+    uint8_t * bufPtr = buffer;
+
+    /* Dummy network inteface - pointer to pointer to buffer. */
+    void * pNetworkInterface = ( void * ) &bufPtr;
+
+    buffer[ 0 ] = 0x20; /* CONN ACK */
+    buffer[ 1 ] = 0x02; /* Remaining length. */
+
+    status = IotMqtt_GetIncomingMQTTPacketTypeAndLength( &mqttPacket, _getNextByte, pNetworkInterface );
+    TEST_ASSERT_EQUAL_INT( IOT_MQTT_SUCCESS, status );
+    TEST_ASSERT_EQUAL_INT( 0x20, mqttPacket.type );
+    TEST_ASSERT_EQUAL_INT( 0x02, mqttPacket.remainingLength );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Tests that IotMqtt_DeserializeResponse works as intented.
+ * to @ref mqtt_function_deserializeresponse.
+ */
+TEST( MQTT_Unit_API, DeserializeResponseChecks )
+{
+    IotMqttPacketInfo_t mqttPacketInfo;
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+    uint8_t buffer[ 10 ];
+
+    /* Verify parameters */
+    status = IotMqtt_DeserializeResponse( NULL );
+    TEST_ASSERT_EQUAL_INT( IOT_MQTT_BAD_PARAMETER, status );
+
+    memset( ( void * ) &mqttPacketInfo, 0x00, sizeof( mqttPacketInfo ) );
+
+    mqttPacketInfo.type = 0x01;
+    mqttPacketInfo.pRemainingData = buffer;
+    status = IotMqtt_DeserializeResponse( &mqttPacketInfo );
+    TEST_ASSERT_EQUAL_INT( IOT_MQTT_BAD_PARAMETER, status );
+
+    /* Good case succeeds - Test for CONN ACK */
+    /* Set conn ack variable portion */
+    buffer[ 0 ] = 0x00;
+    buffer[ 1 ] = 0x00;
+    /* Set type, remaining length and remaining data. */
+    mqttPacketInfo.type = 0x20;         /* CONN ACK */
+    mqttPacketInfo.pRemainingData = buffer;
+    mqttPacketInfo.remainingLength = 0x02; /* CONN ACK Remaining Length. */
+    status = IotMqtt_DeserializeResponse( &mqttPacketInfo );
+    TEST_ASSERT_EQUAL_INT( IOT_MQTT_SUCCESS, status );
+
+    /* TODO add test for each acks */
```

Comment:
Can we add these?  If not, please remove `TODO`.

Reply:
I'll remove TODO for now I'll add more tests  later.  Since we are not changing library AppSec should not be affected when I add more unit tests.